lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -2850,6 +2850,18 @@ LogicalResult SliceLibOp::verify() {
   return success();
 }
 
+// Undef op
+SmallVector<StringRef> UndefLibOp::portNames() { return {"out"}; }
+SmallVector<Direction> UndefLibOp::portDirections() { return {Output}; }
```

Comment:
ubernit: i know this is not the style used in the rest of the Calyx ops, but i would normally place such simple functions in the ODS `extraClassDeclarations` segment.

Reply:
I noticed the lack of consistency on this too! Seems like a good idea to keep things in the `td` file as much as possible.